### PR TITLE
Additional Doc Updates

### PR DESCRIPTION
**Pull request by kbarnhart**
This is the another PR updating the docs motivated by:
a) My desire to update how Landlab presents citations
b) Issue #1108 

- I've added a References section all components. 
- Where a software citation exits, now all components have one in _cite_as
- I update Component to have a public property called cite_as... it just prints _cite_as. 
- I expanded the section in the docs that describes how to use the citation registry, and gave an example of the prior functionality. 
- I removed the table of component citations (since now each component lists citations in a reference section).
- I added a page describing the package organization
- I updated the main index.rst to be more in line with the updated Readme that is currently on #1112 
- In some places we had parameters high up in component documentation, I moved it to after the init where I noticed it. 

~This also does what PR #1116 does. But its buried and it changes @nathanlyons 's component (a tiny tiny bit), so I want to make sure that @nathanlyons thinks that is OK (or says its not and I remove it here too). Ideally this is OK with @nathanlyons, we merge #1116 and then merge this.~ Update. As per conversation with @nathanlyons I've removed all the changes to `landlab/components/species_evolution\` and `tests\components\species_evolution` since @nathanlyons likes these changes, but thinks they need to be dealt with in a different order due to local changes he has made.

### CODE REVIEW

**Comment by kbarnhart** [comment]
@mcflugen @gregtucker @landlab/core any comments on these doc updates. I'd like to merge them (post comments) so I can move forward with the changes to the Readme.md's in #1112 for the GroundwaterDupuitPercolator JOSS review.

File: docs/source/citation_registry.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                  year={2017},
+                  publisher={Copernicus GmbH}
+                  }
+
+In addition, you can use the "citation registry" to make a .bib file of all
+citations you've used in script.
+
+.. code-block:: python
+
+    import landlab
```
Comment (nathanlyons) [inline docs/source/citation_registry.rst]: In this branch, this works for me in 
```python
import landlab
w = landlab.registry.format_citations()
with open("citations.bib", "w") as f:
    f.write(w)
```
Although, it breaks with a component, e.g.:
```python
import landlab
from landlab.components import LinearDiffuser

mg = landlab.RasterModelGrid((3,3))
mg.add_ones('topographic__elevation', at='node')
ed = LinearDiffuser(mg)

w = landlab.registry.format_citations()
with open("citations.bib", "w") as f:
    f.write(w)
```
I get this error
```python
  File "/Users/njlyons/repos/landlab/landlab/landlab/core/messages.py", line 139, in indent_and_wrap
    first_line, the_rest = [lines[0].strip()], lines[1:]

IndexError: list index out of range
```

**Comment by kbarnhart** [comment]
@nathanlyons thanks for the comments. I'll address them.

File: docs/source/index.rst [hunk 2]
```diff
@@ -6,17 +6,29 @@
...
+  a common interface and can be combined based on a users needs.
+* Utilities which support general numerical methods, file input/output, and
+  visualization.
+
+In addition Landlab contains a set of Jupyter notebook tutorials providing
+introduction to core concepts and examples of use.
+
+Landlab was desiged for disciplines that quantify Earth surface dynamics such
+as geomorphology, hydrology, glaciology, and stratigraphy, it can also be used
+in related fields. Scientists who want to use a this type of model often build
```
[the remark this answers is not in the document]
Comment (kbarnhart) [reply docs/source/index.rst]: fixed. :tada: to your attention to detail!

File: docs/source/index.rst [hunk 3]
```diff
@@ -100,50 +113,35 @@ BibTeX format:
...
+`GitHub Issue <https://github.com/landlab/landlab/issues>`_
 
-  # the code of your model built with landlab goes here, then add a call to
+To keep in touch with the latest Landlab news:
 
-  landlab.registry.format_citations() # will produce a Bibtex-formatted
-                                      # citations for all Landlab components
-                                      # that you currently have instantiated.
+-  Get the `Landlab Lookout Newsletter <https://github.us18.list-manage.com/subscribe?u=2db7cea82e3ea40fcf4c91247&id=b9bad233c7>`_
+-  Landlab is on `Twitter <https://mobile.twitter.com/landlabtoolkit>`_!
```
[the remark this answers is not in the document]
Comment (kbarnhart) [reply docs/source/index.rst]: good point. fixed.

File: landlab/components/hack_calculator/hack_calculator.py [hunk 4]
```diff
@@ -154,6 +154,17 @@ class HackCalculator(Component):
...
     239                39.0  2160000.0  3000.0  2894.22
     240                39.0    10000.0     0.0    71.61
+
+    References
+    ----------
+    **Required Software Citation(s) Specific to this Component**
+
+    None Listed
+
+    **Additional References**
```
[the remark this answers is not in the document]
Comment (kbarnhart) [reply landlab/components/hack_calculator/hack_calculator.py]: Good point. I've added it and the DOI.

File: docs/source/citation_registry.rst (hunk 1 above)
Comment (kbarnhart) [reply docs/source/citation_registry.rst]: @nathanlyons I've addressed this by removing these examples and replacing them with the docstrings for the `landlab._registry` file. These were not represented in the docs... and are actually tested. I think that is a better solution. 

Thanks for testing these!

**Comment by kbarnhart** [comment]
any other comments from @mcflugen @gregtucker or @landlab/core before I merge this?
thanks @nathanlyons for your comments (👍 )

File: docs/source/index.rst [hunk 5]
```diff
@@ -6,17 +6,29 @@
...
-earth surface dynamics such as geomorphology, hydrology, glaciology, and
-stratigraphy, it can also be used in related fields.
-
-Landlab provides components to compute flows (such as water, sediment,
-glacial ice, volcanic material, or landslide debris) across a gridded
-terrain. With its robust, reusable components, Landlab allows scientists
-to quickly build landscape model experiments and compute mass balance
-across scales. Watch the webinar `Landlab Toolkit Overview <https://csdms.colorado.edu/wiki/Presenters-0407>`_ at CSDMS to learn more.
+Landlab is an open-source Python-language package for numerical modeling of
+Earth surface dynamics. it contains
```
Comment (gregtucker) [inline docs/source/index.rst]: It contains:

File: docs/source/index.rst [hunk 6]
```diff
@@ -6,17 +6,29 @@
...
+
+* A gridding engine which represents the model domain. Regular and irregular
+  grids are supported.
+* A library of process components, each of which represents a physical process
+  (e.g., generation of rain, erosion by flowing water). These components have
+  a common interface and can be combined based on a users needs.
```
Comment (gregtucker) [inline docs/source/index.rst]: user's

File: docs/source/index.rst [hunk 7]
```diff
@@ -6,17 +6,29 @@
...
-
-Landlab provides components to compute flows (such as water, sediment,
-glacial ice, volcanic material, or landslide debris) across a gridded
-terrain. With its robust, reusable components, Landlab allows scientists
-to quickly build landscape model experiments and compute mass balance
-across scales. Watch the webinar `Landlab Toolkit Overview <https://csdms.colorado.edu/wiki/Presenters-0407>`_ at CSDMS to learn more.
+Landlab is an open-source Python-language package for numerical modeling of
+Earth surface dynamics. it contains
+
+* A gridding engine which represents the model domain. Regular and irregular
```
Comment (gregtucker) [inline docs/source/index.rst]: that

File: docs/source/index.rst [hunk 8]
```diff
@@ -6,17 +6,29 @@
...
+  grids are supported.
+* A library of process components, each of which represents a physical process
+  (e.g., generation of rain, erosion by flowing water). These components have
+  a common interface and can be combined based on a users needs.
+* Utilities which support general numerical methods, file input/output, and
```
Comment (gregtucker) [inline docs/source/index.rst]: that

File: docs/source/index.rst [hunk 9]
```diff
@@ -6,17 +6,29 @@
...
+  visualization.
+
+In addition Landlab contains a set of Jupyter notebook tutorials providing
+introduction to core concepts and examples of use.
```
Comment (gregtucker) [inline docs/source/index.rst]: an introduction

File: docs/source/index.rst [hunk 10]
```diff
@@ -6,17 +6,29 @@
...
+
+Landlab was desiged for disciplines that quantify Earth surface dynamics such
```
Comment (gregtucker) [inline docs/source/index.rst]: dynamics, such ... stratigraphy. It can ...

File: docs/source/index.rst [hunk 11]
```diff
@@ -6,17 +6,29 @@
...
+as geomorphology, hydrology, glaciology, and stratigraphy, it can also be used
+in related fields. Scientists who use this type of model often build
+their own unique model from the ground up, re-coding the basic building blocks
+of their landscape model rather than taking advantage of codes that have
+already been written.
```
Comment (gregtucker) [inline docs/source/index.rst]: add: Landlab saves practitioners from the need for this kind of re-invention by providing standardized components that they can re-use.

File: docs/source/index.rst (hunk 5 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed.

File: docs/source/index.rst (hunk 7 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed.

File: docs/source/index.rst (hunk 9 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed.

File: docs/source/index.rst (hunk 6 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed

File: docs/source/index.rst (hunk 8 above)
Comment (kbarnhart) [reply docs/source/index.rst]: someday I'll figure which/that out... fixed.

File: docs/source/index.rst (hunk 10 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed.

File: docs/source/index.rst (hunk 11 above)
Comment (kbarnhart) [reply docs/source/index.rst]: fixed.

**Comment by kbarnhart** [comment]
@gregtucker I'm making your changes in the Readme.md over on #1112 since some of the same text is present.

**Comment by kbarnhart** [comment]
[Linking to the JOSS Review Issue](https://github.com/openjournals/joss-reviews/issues/1935)

**Pull request merged**